### ISSUE TITLE
csvscrob: wrong date calculation when moving past midnight

### ISSUE THREAD

**Issue by arcctgx** [arcctgx/tagreaders#2]
```
> csvscrob -t "2017-04-29 23:58:45" 01_stream_endless.ogg 
"Lull", "Stream Endless", "Dreamt About Dreaming", "2017-04-30 01:02:45", "", "709
```
This is correctly scrobbled after 4 minutes, but for some reason one hour is added to the time.

**Comment by arcctgx** [comment]
Seems to happen only when DST is active.

Also seems to happen only when the day is specified, I can't reproduce it when I specify only the hour:
```
> csvscrob -t 23:58:45 01_stream_endless.ogg 
"Lull", "Stream Endless", "Dreamt About Dreaming", "2017-05-01 00:02:45", "", "709"
```
In this case day and hour are advanced correctly.

Switching to GMT when performing time calculations seems to solve this, but introduces other problems (and requires using nonportable functions in code).

**Comment by arcctgx** [comment]
Maybe fixed by e1e8431, time zone handling is confusing and I don't know anymore.
